packages/admin/admin/src/dateTime/DatePicker.tsx

```diff
@@ -0,0 +1,190 @@
+    value?: string;
+    onChange?: (date: string | undefined) => void;
+    iconMapping?: {
+        openPicker?: React.ReactNode;
+        readOnly?: React.ReactNode;
+    };
+} & Omit<MuiDatePickerProps<Date, true>, "value" | "onChange" | "slotProps">;
+
+const getDateValue = (value: string | undefined): Date | null => {
+    if (!value) {
+        return null;
+    }
+
+    const date = new Date(value);
+
+    if (isNaN(date.getTime())) {
+        throw new Error(`Invalid date value: ${value}`);
+    }
+
+    return date;
+};
+
+export const Future_DatePicker = (inProps: Future_DatePickerProps) => {
+    const {
+        iconMapping = {},
+        fullWidth,
+        required,
+        slotProps,
+        disabled,
+        value: stringValue,
+        onChange,
+        readOnly,
+        ...restProps
+    } = useThemeProps({
+        props: inProps,
+        name: "CometAdminFutureDatePicker",
+    });
+    const [open, setOpen] = useState(false);
+    const dateValue = getDateValue(stringValue);
+
+    const { openPicker: openPickerIcon = <Calendar color="inherit" />, readOnly: readOnlyIcon = <Lock fontSize="inherit" /> } = iconMapping;
+
+    const clearButtonEndAdornment =
+        !required && !disabled && !readOnly ? (
+            <ClearInputAdornment
+                position="end"
+                hasClearableContent={Boolean(dateValue)}
+                onClick={() => onChange?.(undefined)}
+                {...slotProps?.clearInputAdornment}
+            />
+        ) : null;
+
+    const readOnlyAdornment = readOnly ? (
+        <ReadOnlyAdornment position="end" {...slotProps?.readOnlyAdornment}>
+            {readOnlyIcon}
+        </ReadOnlyAdornment>
+    ) : null;
+
+    const openPickerAdornment = (
+        <OpenPickerAdornment position="start" {...slotProps?.openPickerAdornment}>
+            <OpenPickerButton onClick={() => setOpen(true)} disabled={disabled || readOnly} {...slotProps?.openPickerButton}>
+                {openPickerIcon}
+            </OpenPickerButton>
+        </OpenPickerAdornment>
+    );
+
+    return (
+        <Root
+            enableAccessibleFieldDOMStructure
+            disabled={disabled}
+            readOnly={readOnly}
+            open={open}
+            onOpen={() => setOpen(true)}
+            onClose={() => setOpen(false)}
+            disableOpenPicker={true}
+            value={dateValue}
+            onChange={(date) => onChange?.(date ? getIsoDateString(date) : undefined)}
+            {...slotProps?.root}
+            {...restProps}
+            slotProps={{
+                ...slotProps?.root?.slotProps,
+                textField: {
+                    fullWidth,
+                    required,
+                    ...slotProps?.root?.slotProps?.textField,
+                    InputProps: {
+                        // @ts-expect-error It's unclear in the types which variant of `textField` is used here and we can't pass through the correct generic type to it
```

Comment:
We recently had an issue where ts-expect-error silenced two errors in the same line, while only one should be silenced. Therefore I'm inclined to prefer an as cast over ts-expect-error. Would that be possible here?

Reply:
Good point 👍🏼 

But it turns out, I misunderstood the error 😅 
It's fixed now with [Fix TS error by adding the missing `ownerState` to the props of the `TextField` slot](8c43b360bea1d596870653f1e67b42bf617e0c11)